### PR TITLE
Generalize tutorials for audio and vision

### PR DESCRIPTION
**Pull request by stevhliu**
This PR updates the tutorials to be more generalizable to all modalities. After reading the tutorials, a user should be able to load any type of dataset, know how to index into and slice a dataset, and do the most basic/common type of preprocessing (tokenization, resampling, applying transforms) depending on their dataset.

Other changes include:

- Removed the sections about a dataset's metadata, features, and columns because we cover this in an earlier tutorial about inspecting the `DatasetInfo` through the dataset builder.
- Separate the sharing dataset tutorial into two sections: (1) uploading via the web interface and (2) using the `huggingface_hub` library.
- Renamed some tutorials in the TOC to be more clear and specific.
- Added more text to nudge users towards joining the community and asking questions on the forums.
- If it's okay with everyone, I'd also like to remove the section about loading and using metrics since we have the `evaluate` docs now.

### CODE REVIEW

File: docs/source/access.mdx [hunk 1]
```diff
@@ -1,126 +1,81 @@
...
-    'label': ClassLabel(num_classes=2, names=['not_equivalent', 'equivalent'], names_file=None, id=None),
-    'sentence1': Value(dtype='string', id=None),
-    'sentence2': Value(dtype='string', id=None),
-}
+>>> dataset[0]["text"]
+'the rock is destined to be the 21st century\'s new " conan " and that he\'s going to make a splash even greater than arnold schwarzenegger , jean-claud van damme or steven segal .'
 ```
 
-Return even more specific information about a feature like [`ClassLabel`], by calling its parameters `num_classes` and `names`:
+But it is important to remember that indexing order matters, especially when you're working with large or audio and image datasets. Indexing by the column name returns all the values in the column first, then loads the value at that position. For large datasets, it may be slower to index by the column name first.
```
Comment (mariosasko) [inline docs/source/access.mdx:51]:
```suggestion
But it is important to remember that indexing order matters, especially when working with large audio and image datasets. Indexing by the column name returns all the values in the column first, then loads the value at that position. For large datasets, it may be slower to index by the column name first.
```

File: docs/source/load_hub.mdx [hunk 2]
```diff
@@ -1,103 +1,101 @@
-# Hugging Face Hub
+# Load a dataset from the Hub
 
-Now that you are all setup, the first step is to load a dataset. The easiest way to load a dataset is from the [Hugging Face Hub](https://huggingface.co/datasets). There are already over 6000 datasets in over 100 languages on the Hub. Choose from a wide category of datasets to use for NLP tasks like question answering, summarization, machine translation, and language modeling.
+Finding good high-quality datasets that are reproducible and accessible can be difficult. One of 🤗 Datasets main goal is to provide a simple way to load a dataset of any format or type. The easiest way to get started is to discover an existing dataset on the [Hugging Face Hub](https://huggingface.co/datasets) - a community-driven collection of datasets for tasks in NLP, computer vision, and audio - and use 🤗 Datasets to download and generate the dataset.
```
Comment (mariosasko) [inline docs/source/load_hub.mdx:3]:
```suggestion
Finding high-quality datasets that are reproducible and accessible can be difficult. One of 🤗 Datasets main goals is to provide a simple way to load a dataset of any format or type. The easiest way to get started is to discover an existing dataset on the [Hugging Face Hub](https://huggingface.co/datasets) - a community-driven collection of datasets for tasks in NLP, computer vision, and audio - and use 🤗 Datasets to download and generate the dataset.
```

File: docs/source/load_hub.mdx [hunk 3]
```diff
@@ -1,103 +1,101 @@
...
+        features: ['text', 'label'],
+        num_rows: 1066
+    })
+})
 ```
 
-## Select a split
+## Configurations
+
+Some datasets contain several sub-datasets. For example, the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset has several sub-datasets, each one containing audio data in a different language. These sub-datasets are known as *configurations*, and you must explicitly select one when you load the dataset. If you don't provide a configuration name, 🤗 Datasets will raise a `ValueError` and remind you to choose a configuration.
```
Comment (mariosasko) [inline docs/source/load_hub.mdx:83]:
```suggestion
Some datasets contain several sub-datasets. For example, the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset has several sub-datasets, each one containing audio data in a different language. These sub-datasets are known as *configurations*, and you must explicitly select one when loading the dataset. If you don't provide a configuration name, 🤗 Datasets will raise a `ValueError` and remind you to choose a configuration.
```

File: docs/source/upload_dataset.mdx [hunk 4]
```diff
@@ -1,143 +1,133 @@
...
+```bash
 huggingface-cli login
 ```
 
-2. Upload the dataset:
+3. Use the [`push_to_hub()`](https://huggingface.co/docs/datasets/master/en/package_reference/main_classes#datasets.DatasetDict.push_to_hub) function to help you add, commit, and push a file to your repository:
 
 ```py
 >>> from datasets import load_dataset
+>>> from huggingface_hub import Repository
```
Comment (mariosasko) [inline docs/source/upload_dataset.mdx:94]: Unused import:

```suggestion
```

File: docs/source/use_dataset.mdx [hunk 5]
```diff
@@ -1,137 +1,188 @@
...
-}
+>>> from transformers import DataCollatorWithPadding
+
+>>> data_collator = DataCollatorWithPadding(tokenizer=tokenizer, return_tensors="tf")
+>>> tf_dataset = dataset.to_tf_dataset(
+...     columns=["input_ids", "token_type_ids", "attention_mask"],
+...     label_cols=["labels"],
+...     batch_size=2,
+...     collate_fn=data_collator,
+...     shuffle=True)
```
Comment (mariosasko) [inline docs/source/use_dataset.mdx:91]: Formatting:

```suggestion
>>> tf_dataset = dataset.to_tf_dataset(
...     columns=["input_ids", "token_type_ids", "attention_mask"],
...     label_cols=["labels"],
...     batch_size=2,
...     collate_fn=data_collator,
...     shuffle=True
... )
```

**Review by mariosasko — COMMENTED** [review]
Looks good already. 

Maybe instead of removing the metrics docs abruptly, we can deprecate them for now (with a warning?) and redirect users to the `evaluate` docs.

Some nits:

**Review by lhoestq — APPROVED** [review]
Awesome ! And +1 about deprecating metrics instead of removing them, and redirecting to `evaluate`

File: docs/source/metrics.mdx [hunk 6]
```diff
@@ -1,5 +1,11 @@
 # Evaluate predictions
 
+<Tip warning={true}>
+
+Metrics will soon be deprecated in 🤗 Datasets. To learn more about how to use metrics, take a look at our newest library 🤗 [Evaluate](https://huggingface.co/docs/evaluate/index)! In addition to metrics, we've also added more tools for evaluating models and datasets.
+
+</Tip>
```
Comment (lhoestq) [inline docs/source/metrics.mdx:7]: Sounds good to me, WDYT @mariosasko ?

Comment (mariosasko) [reply docs/source/metrics.mdx:7]: Yes, looks good!
